docs/source/conf.py

```diff
@@ -315,39 +315,64 @@ def package_list_from_file(file):
-        for part in info["fullname"].split("."):
-            obj = getattr(obj, part)
-        fname = inspect.getsourcefile(obj)
-        # https://github.com/rtfd/readthedocs.org/issues/5735
-        if any(s in fname for s in ("readthedocs", "rtfd", "checkouts")):
-            # /home/docs/checkouts/readthedocs.org/user_builds/pytorch_lightning/checkouts/
-            #  devel/pytorch_lightning/utilities/cls_experiment.py#L26-L176
-            path_top = os.path.abspath(os.path.join("..", "..", "..", ".."))
-            fname = os.path.relpath(fname, start=path_top)
-        else:
-            # Local build, imitate master
-            fname = "master/" + os.path.relpath(fname, start=os.path.abspath(".."))
-        source, lineno = inspect.getsourcelines(obj)
-        return fname, lineno, lineno + len(source) - 1
+    # try to find the file and line number, based on code from numpy:
+    # https://github.com/numpy/numpy/blob/master/doc/source/conf.py#L424
 
     if domain != "py" or not info["module"]:
         return None
+
+    module_name = info["module"]
+    full_name = info["fullname"]
+
+    sub_module = sys.modules.get(module_name)
+    if sub_module is None:
+        return None
+
+    obj = sub_module
+    for part in full_name.split("."):
+        try:
+            obj = getattr(obj, part)
+        except Exception:
+            return None
+
+    # strip decorators, which would resolve to the source of the decorator
+    # possibly an upstream bug in getsourcefile, bpo-1764286
+    try:
+        unwrap = inspect.unwrap
+    except AttributeError:
+        pass
+    else:
+        obj = unwrap(obj)
+
+    try:
+        file_name = inspect.getsourcefile(obj)
+    except Exception:
+        file_name = None
+    if not file_name:
+        return None
+
+    module = inspect.getmodule(obj)
+    if module is not None and not module.__name__.startswith("torchmetrics"):
+        return None
+
     try:
-        filename = "%s#L%d-L%d" % find_source()
-    except Exception:  # todo: specify the exception
-        filename = info["module"].replace(".", "/") + ".py"
-    # import subprocess
-    # tag = subprocess.Popen(['git', 'rev-parse', 'HEAD'], stdout=subprocess.PIPE,
-    #                        universal_newlines=True).communicate()[0][:-1]
-    branch = filename.split("/")[0]
-    # do mapping from latest tags to master
-    branch = {"latest": "master", "stable": "master"}.get(branch, branch)
-    filename = "/".join([branch] + filename.split("/")[1:])
-    return f"https://github.com/{github_user}/{github_repo}/blob/{filename}"
+        source, line_number = inspect.getsourcelines(obj)
+    except Exception:
+        source, line_number = None, None
+    file_name = os.path.relpath(file_name, start=os.path.dirname(torchmetrics.__file__))
+
+    if line_number and source:
+        line_str = "#L%d-L%d" % (line_number, line_number + len(source) - 1)
+    else:
+        line_str = ""
+
+    if "dev" in torchmetrics.__version__:
```

Comment:
Yes. Should we just replace with the following code? 
```python
if any(s in torchmetrics.__version__ for s in ("dev", "rc")):
    version_str = "master"
else:
    version_str = f"v{torchmetrics.__version__}"
```
The documentation generated with `rc` and `dev` is always the documentation of the `latest` version on ReadTheDocs right? In this case I think it would be fine if the docs for `rc` and `dev` versions link to the code from `master`.

Reply:
Should I push that change? @Borda

Reply:
could you, or send it in another pr?

Reply:
Done. I tested locally with `torch.__version__ = "0.11.0"`, `torch.__version__ = "0.12.0rc"` and `torch.__version__ = "0.12.0dev"` and it seems to work in all cases.